src/components/HoverCard/HoverCardOnUsername.tsx

```diff
@@ -0,0 +1,168 @@
+import { useState } from 'react';
+import colors from 'components/core/colors';
+import InteractiveLink from 'components/core/InteractiveLink/InteractiveLink';
+import { BaseM, TitleM } from 'components/core/Text/Text';
+import styled, { css, keyframes } from 'styled-components';
+import { graphql, useFragment } from 'react-relay';
+import { HoverCardOnUsernameFragment$key } from '__generated__/HoverCardOnUsernameFragment.graphql';
+import Markdown from 'components/core/Markdown/Markdown';
+import unescape from 'lodash/unescape';
+import FollowButton from 'components/Follow/FollowButton';
+import transitions, {
+  ANIMATED_COMPONENT_TRANSLATION_PIXELS_LARGE,
+} from 'components/core/transitions';
+import { HoverCardOnUsernameFollowFragment$key } from '__generated__/HoverCardOnUsernameFollowFragment.graphql';
+import { useLoggedInUserId } from 'hooks/useLoggedInUserId';
+
+type Props = {
+  username: string;
+  userRef: HoverCardOnUsernameFragment$key;
+  queryRef: HoverCardOnUsernameFollowFragment$key;
+};
+
+export default function HoverCardOnUsername({ username, userRef, queryRef }: Props) {
+  const [isHovering, setIsHovering] = useState(false);
+
+  const user = useFragment(
+    graphql`
+      fragment HoverCardOnUsernameFragment on GalleryUser {
+        username
+        bio
+        galleries @required(action: THROW) {
+          collections {
+            dbid
+            name
+          }
+        }
+        ...FollowButtonUserFragment
+      }
+    `,
+    userRef
+  );
+
+  const query = useFragment(
+    graphql`
+      fragment HoverCardOnUsernameFollowFragment on Query {
+        ...FollowButtonQueryFragment
+        ...useLoggedInUserIdFragment
+      }
+    `,
+    queryRef
+  );
+
+  const totalCollections = user?.galleries[0]?.collections?.length || 0;
+
+  const handleMouseEnter = () => {
+    setIsHovering(true);
+  };
+
+  const handleMouseLeave = () => {
+    setIsHovering(false);
+  };
+
+  const loggedInUserId = useLoggedInUserId(query);
+  const isLoggedIn = !!loggedInUserId;
+
+  return (
+    <StyledContainer onMouseEnter={handleMouseEnter} onMouseLeave={handleMouseLeave}>
+      <StyledLinkContainer>
+        <InteractiveLink to={`/${username}`}>{username}</InteractiveLink>
+      </StyledLinkContainer>
+
+      <StyledCardWrapper isHovering={isHovering}>
```

Comment:
does the hover behavior work if we only displayed `StyledCardWrapper` if `isHovering` is true?
like this:
{isHovering && <StyledCardWrapper>...</> }
It looks like we render all the card wrappers in the feed and set opacity to 1 if the user hovers on one, but maybe it's possible to make it so that we only render a single card wrapper when the user hovers on a name?

Reply:
Yes it works. We can do that way too. 

Might need to change a little bit on how the animation appeared

Reply:
Updated it 👌